nixos/modules/config/system-environment.nix

```diff
@@ -75,13 +70,20 @@ in
       let
         suffixedVariables =
           flip mapAttrs cfg.profileRelativeSessionVariables (envVar: suffixes:
-            flip concatMap pamProfiles (profile:
+            flip concatMap cfg.profiles (profile:
               map (suffix: "${profile}${suffix}") suffixes
             )
           );
 
+        # We're trying to use the same syntax for PAM variables and env variables.
+        # That means we need to map the env variables that people might use to their
+        # equivalent PAM variable.
+        # Note: PAM_USER is a PAM_ITEM, HOME is an environment variable, they have 
```

Comment:
So actually '@{HOME}` is special, and there are important reasons for preferring it which are outlined in `man pam_env.conf`:

> (Possibly non-existent) environment variables may be used in values using the ${string} syntax and (possibly non-existent)
> PAM_ITEMs as well as HOME and SHELL may be used in values using the @{string} syntax. Both the $ and @ characters can be
> backslash escaped to be used as literal values values can be delimited with "", escaped " not supported. Note that many
> environment variables that you would like to use may not be set by the time the module is called. For example, ${HOME} is
> used below several times, but many PAM applications don't make it available by the time you need it. The special variables
> @{HOME} and @{SHELL} are expanded to the values for the user from his passwd entry.

I remember running into problems when using $HOME instead of @{HOME} (as the quoted bit above says, HOME often isn't set by the time it's needed) but bouncing through commit logs on various machines I'm not finding evidence of such a (mis-)adventure.  It may have been related to sshd/sudo, but regardless I am inclined to suggest we continue to prefer @{HOME} in pam-processed bits.  That said I'm still sorting through this and similar recent changes and don't quite have it all in my head just yet :).

Reply:
If the docs say that, it's still a fairly easy change to revert
```diff
- replaceEnvVars = replaceStrings ["$HOME" "$USER"] ["\${HOME}" "@{PAM_USER}"];
+ replaceEnvVars = replaceStrings ["$HOME" "$USER"] ["@{HOME}" "@{PAM_USER}"];
```

and wasn't the motivation for this change anyways.

Reply:
@dtzWill btw you'd be an excellent reviewer for https://github.com/NixOS/nixpkgs/pull/70430 (I think anyways) :smile:

Reply:
Ah, here we go.  Test is `sudo env`-- with the latest changes it's clear everything is wrong (HOME isn't set as described above, so all the variables are borked), reverting to use @{HOME} fixes them again.

Please confirm/test and let me know what you find.

Possibly part of what's going on with #70418, but don't have a full story explaining that behavior just yet.

Reply:
Ahh I see. Testing this change now. PR incoming since it should be completely fine.

Reply:
@dtzWill Yep `sudo env` now looks right.

Reply:
:heart: thanks for quick response, will take a look! And yes I don't think this will be hard to fix or require big reverts or anything :).